public/test/helpers/TestProvider.tsx

```diff
@@ -20,6 +20,8 @@ export interface Props {
 
 /**
  * Wrapps component in redux store provider, Router and GrafanaContext
+ *
+ * @deprecated Use `@grafana/test-utils` `render` method instead
```

Comment:
(as with everything in this PR) Up for discussion - maybe this shouldn't be deprecated but instead this provider now adds all of the wrappers as in the new test-utils render method? 
I would have added all of them in here but as there are lots of uses of this `TestProvider` already, we might end up with duplication of Provider wrappers.

Or regardless, perhaps I should move the `Wrapper` from test utils into a separate file anyway?

Reply:
It's difficult to do a full benchmark to compare, as it requires moving nearly every test over, but anecdotally based on some of the tests in https://github.com/grafana/grafana/pull/85673, it doesn't seem to have a big impact. 

I think the benefit though is that the tests will render with everything that they would otherwise be wrapped in in a real usage. It's nice to be able to update your test helper method and know that every test using it will also be testing using that

However, this is obviously overkill for most of the design library/core components that don't need any providers, so I could export the "raw" render method from RTL as well so the methods could still be used via `@grafana/test-utils`? What do you think?